word-guess.rb

```diff
@@ -0,0 +1,156 @@
+# This is a word guessing game (by Erika & Ari)
+# Guess the word before all the petals fall
+# use one class for game functionality
+require 'colorize'
+require 'spicy-proton'
+
+
+class Game
+
+ def initialize(word)
+   @correct_word_array = word.downcase.split("")
+   # This take the random word, downcase, and split letters into it's own indexes
+   # or elements (interchangable right?)
+   @correct_guess = word.gsub(/./ , "_").split("")
+# gsub method lets us sub in "-". "/./" is for any characters.
+# split method puts "-" into own index or makes them their own elements
+   @no_match_guesses = []
+   @failed_attempt = 0
+ end
+
+ def pedal_remove
+   pedal_remove = [
+     '''
+       (@)(@)(@)(@)(@)
+          ,\,\,|,/,/,
+            _\|/_
+           |_____|
+            |   |
+            |___|'''.cyan,'''
+
+        (@)(@)(@)(@)
+        ,\,\,|,/,/,
+            _\|/_
+           |_____|
+            |   |
+            |___|'''.green,'''
+
+          (@)(@)(@)
+         ,\,\,|,/,/,
+            _\|/_
+           |_____|
+            |   |
+            |___|'''.yellow,'''
+
+           (@)(@)
+         ,\,\,|,/,/,
+            _\|/_
+           |_____|
+            |   |
+            |___|'''.blue,'''
+
+              (@)
+          ,\,\,|,/,/,
+             _\|/_
+            |_____|
+             |   |
+             |___|'''.red]
+ end
+
+ def promptUser
+   unless (@correct_guess.include?("_") )
+     puts "\nYOU WIN! THE WORD WAS: #{@correct_word_array.join("")}".green
+     # if user guess all the letters correctly
+      return
+   end
+
+   # ask the user to pick a letter
+   puts "\nGUESS A LETTER!".green
+   guess = gets.chomp.downcase
+
+   # prints guesses to user followed by comma
+   puts "YOUR GUESSES SO FAR: #{@no_match_guesses.join(', ')}"
+
+   # let the user know if they can't have more than one letter
+     if (guess.length > 1)
+       puts "\nYOU HAVE TO GUESS ONE LETTER AT A TIME.".red
+       return promptUser
+     end
+
+   # let the user know it has to be a letter
+   unless (guess.match(/[a-zA-Z]/))
+     puts "\nSORRY. MUST BE A LETTER".red
+     return promptUser
+   end
+
+   # if letter guessed is already picked
+   if (@no_match_guesses.include?(guess))
+     puts "\nYOU'VE ALREADY GUESSED THIS LETTER: #{guess}".red
+     pedal_remove
+     return promptUser
+   end
+   # track of all the letters guessed
+   @no_match_guesses.push(guess)
+
+   # if the users letter is in the word
+   if (@correct_word_array.include?(guess)) #from index [0] to last index of word
+     0.upto(@correct_word_array.length-1) { |index|
+       if (@correct_word_array[index] == guess)
```

Comment:
I agree that this logic is a little dense. One observation is that we don't need the check on line 96, because if the correct word does not include the guess, then the if statement on line 98 will never go off.

We could simplify the whole section by rewriting it as:

```ruby
letter_found = false
@correct_word_array.each_with_index do |letter, index|
  if letter == guess
    @correct_guess[index] = guess
    letter_found = true
  end
end

if letter_found
  puts "SECERT WORD: #{@correct_guess.join('  ')}"
  promptUser # return to the prompt method to guess another letter
else
  # ... the rest of this method, starting on line 105 ...
end
```